vms/platformvm/state/state.go

```diff
@@ -831,6 +833,64 @@ func (s *state) DeleteExpiry(entry ExpiryEntry) {
 	s.expiryDiff.DeleteExpiry(entry)
 }
 
+func (s *state) GetCurrentValidatorSet(ctx context.Context, subnetID ids.ID) (map[ids.ID]*validators.GetCurrentValidatorOutput, uint64, error) {
+	result := make(map[ids.ID]*validators.GetCurrentValidatorOutput)
+	for _, staker := range s.currentStakers.validators[subnetID] {
+		if err := ctx.Err(); err != nil {
+			return nil, 0, err
+		}
+		validator := staker.validator
+		result[validator.TxID] = &validators.GetCurrentValidatorOutput{
+			ValidationID:   validator.TxID,
+			NodeID:         validator.NodeID,
+			PublicKey:      validator.PublicKey,
+			Weight:         validator.Weight,
+			StartTime:      uint64(validator.StartTime.Unix()),
+			SetWeightNonce: 0,
+			IsActive:       true,
+			IsSoV:          false,
+		}
+	}
+
+	// this is L1 subnet, fetch all validation IDs from subnetIDNodeIDDB
+	// with subnetID as prefix
```

Comment:
This comment seems out of date. We don't know if this is an L1 here or not (but we don't care)